benchmark/single-source/StringComparison.swift.gyb

```diff
@@ -0,0 +1,258 @@
+//===--- StringComparison.swift -------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+% # Ignore the following warning. This _is_ the correct file to edit.
+////////////////////////////////////////////////////////////////////////////////
+// WARNING: This file is manually generated from .gyb template and should not
+// be directly modified. Instead, make changes to StringComparison.swift.gyb and run
+// scripts/generate_harness/generate_harness.py to regenerate this file.
+////////////////////////////////////////////////////////////////////////////////
+
+//
+// Test String iteration performance over a variety of workloads, languages,
+// and symbols.
+//
+
+import TestsUtils
+
+extension String {
+  func lines() -> [String] {
+    return self.split(separator: "\n").map { String($0) }
+  }
+}
+
+% Names = ["ascii", "latin1", "fastPrenormal", "slowerPrenormal", "nonBMPSlowestPrenormal", "emoji", "abnormal", "zalgo"]
+
+public let StringComparison = [
+% for Name in Names:
+  BenchmarkInfo(
+    name: "StringComparison_${Name}",
+    runFunction: run_StringComparison_${Name},
+    tags: [.validation, .api, .String]),    
+% end # Names
+]
+    
+% for Name in Names:
+  @inline(never)
+  public func run_StringComparison_${Name}(_ N: Int) {
+    var count = 0
+    let workload = Workload.${Name}
+    let tripCount = workload.tripCount
+    let payload = workload.payload
+    for _ in 1...tripCount*N {
+      for s1 in payload {
+        for s2 in payload {
+          let cmp = s1 < s2
+          count += cmp ? 1 : 0
```

Comment:
These days, the benchmark suite has a `blackHole` function (inline never in another module) that you can also feed results into. That's an additional guarantee that the entire benchmark won't be trivialized by a better optimizer. (This is why many benchmarks followed the pattern of an inline(never) helper that returned a result derived from the inner loop, though that wasn't always perfect and `blackHole` is better).